libfoundation/src/system-dylib-mac.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright (C) 2016 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include "system-private.h"
+
+#include <foundation-auto.h>
+#include <mach-o/dyld.h>
+
+void *_MCSDylibLoadModule(MCStringRef p_path)
+{
+	MCAutoStringRefAsSysString t_filename_sys;
+	/* UNCHECKED */ t_filename_sys.Lock(p_path);
+	
+	void* t_result;
+	t_result = (void *)NSAddImage(*t_filename_sys, NSADDIMAGE_OPTION_RETURN_ON_ERROR | NSADDIMAGE_OPTION_WITH_SEARCHING);
+	if (t_result != nil)
+		return t_result;
+	
+	CFURLRef t_url;
```

Comment:
I think this repeats the same bug as the existing MacOS dylib code - it overloads the object handle as being both a `NSImage*` and `CFBundleRef` which will lead to crashes when passed to the wrong function. Do we actually need both the NS code and the CF code?